### ISSUE TITLE
**General contributions**

### ISSUE THREAD

**Issue by spirosmaggioros** [CSRT-NTUA/AlgoPlus#56]
Please be free to contribute whatever you think is useful. Don't only feel the need to contribute if there are issues open :)

**Comment by aveldan** [comment]
I was thinking of implementing a level-order traversal for the tree, I see it's not done yet. If that is okay I can start working on it.

**Comment by spirosmaggioros** [comment]
Yes you can do that, i had one for every tree, but i deleted it because it's not a known algorithm, it's just something we do to solve problems. But yes, you can implement one for every tree header class and then i will create an ostream operator to print the trees using the level order traversal.

**Comment by aveldan** [comment]
I was thinking of implementing Red Black tree, if no one is working on this right now. I can raise an issue and work on it. That include the tree implementation, visualization, tests and examples.

**Comment by spirosmaggioros** [comment]
If you see, i lately added a 2-3-4 class and after that i was going to add a red black tree class as it's the only basic tree structure that is missing, though it has a very difficult implementation. I believe it's the second most difficult structure i've ever implemented, but yes, you can do that, but then i have to check it a lot myself to be sure. You'll save me a lot of time by adding it yourself if you think you can!

**Comment by aveldan** [comment]
I know the concept but have never implemented it in a full scale like this, but yeah I am up for the challenge. 
I will put in a issue, please assign it to be. I will work on it this week.

> but then i have to check it a lot myself to be sure
Yes sure, that is what I expect.

> You'll save me a lot of time by adding it yourself if you think you can!
Happy to help.
